### PR TITLE
[mlir][SparseTensor] Add missing dependent dialect to pass

### PR DESCRIPTION
**Pull request by matthias-springer**
This commit fixes the following error when stopping the sparse compiler pipeline after bufferization (e.g., with `test-analysis-only`):

```
LLVM ERROR: Building op `vector.print` but it isn't known in this MLIRContext: the dialect may not be loaded or this operation hasn't been added by the dialect. See also https://mlir.llvm.org/getting_started/Faq/#registered-loaded-dependent-whats-up-with-dialects-management
```

### CODE REVIEW

**Comment by joker-eph** [comment]
Can you add a test for this?

File: mlir/test/Integration/Dialect/SparseTensor/CPU/sparse_block3d.mlir [hunk 1]
```diff
@@ -30,6 +30,10 @@
 // Do the same run, but now with direct IR generation and VLA vectorization.
 // RUN: %if mlir_arm_sve_tests %{ %{compile_sve} | %{run_sve} | FileCheck %s %}
 
+// Make sure that test-bufferization-analysis-only works. This option is useful
```
Comment (aartbik) [inline mlir/test/Integration/Dialect/SparseTensor/CPU/sparse_block3d.mlir:33]: The way it was phrased looked initially like a TODO to me, but then I realized it was simply tested.

Can we rephrase this as

Test that ....

**Review by aartbik — APPROVED** [review]
